docs/otoc/otoc_example.ipynb

```diff
@@ -0,0 +1,761 @@
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "TNPdutdoRM9Q"
+   },
+   "source": [
+    "Now we create all OTOC circuits for the experiment, varying the number of cycles and the butterfly qubit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "id": "mdARggzhzoHp"
+   },
+   "outputs": [],
+   "source": [
+    "np.random.seed(0)  # Random seed for OTOC circuits.\n",
+    "\n",
+    "circuit_list = []\n",
+    "for i in range(num_trials):\n",
+    "    circuits_i = []\n",
+    "\n",
+    "    for cycle in cycles:\n",
+    "        circuits_ic = []\n",
+    "  \n",
+    "        for k, q_b in enumerate(qubits[1:]):\n",
+    "            circs = otoc.build_otoc_circuits(\n",
+    "                qubits,\n",
+    "                ancilla,\n",
+    "                cycle,\n",
+    "                int_sets,\n",
+    "                forward_ops=forward_ops,\n",
+    "                reverse_ops=reverse_ops,\n",
+    "                butterfly_qubits=q_b,\n",
+    "                cycles_per_echo=2,\n",
+    "                sq_gates=np.random.choice(8, (num_qubits, max(cycles))),\n",
+    "                use_physical_cz=True,\n",
+    "            )\n",
+    "\n",
+    "            # If q_b is the measurement qubit, add both the normalization \n",
+    "            # circuits and the circuits with X as the butterfly operator. \n",
+    "            # Otherwise only add circuits with Y being the butterfly operator.\n",
+    "            if k == 0:\n",
+    "                circuits_ic.extend(circs.butterfly_I)\n",
+    "                circuits_ic.extend(circs.butterfly_X)\n",
+    "            else:\n",
+    "                circuits_ic.extend(circs.butterfly_Y)\n",
+    "\n",
+    "        circuits_i.append(circuits_ic)\n",
+    "    circuit_list.append(circuits_i)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "I3jboTmsHM3j"
+   },
+   "source": [
+    "The `circuit_list` is a three-dimensional array where the first axis corresponds to the trial, the second axis corresponds to the number of cycles, and the third axis corresponds the butterfly operator / qubit ($Q_b$)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Lf8B1rUc-qxE"
+   },
+   "source": [
+    "## Run the circuits"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "tvEfmDVrOA1T"
+   },
+   "source": [
+    "Now we run the circuits and process the results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {
+    "id": "zRsJMaG3EkgD"
+   },
+   "outputs": [],
+   "source": [
+    "# Compile to sqrt_iswap.\n",
+    "for i, circuits_trial in enumerate(circuit_list):\n",
+    "    for j, circuits_cycle in enumerate(circuits_trial):\n",
+    "        for k, circuit_otoc in enumerate(circuits_cycle):\n",
+    "            circuit_list[i][j][k] = cg.optimized_for_sycamore(circuit_otoc, optimizer_type=\"sqrt_iswap\")"
```

Comment:
this takes longer than expected on my machine. maybe `print('.', end='')` just so there's some indication of progress

Reply:
this actually takes quite a long time. maybe we can throw multiprocessing at it or have a real progress bar?

Reply:
Added a progress bar. Compiling is kind of a hack around #189 - ideally I think the desired behavior is that each circuit already has the right gates and no compiling is necessary.

Reply:
ok thanks for the additional information